src/libAtomVM/port.c

```diff
@@ -0,0 +1,94 @@
+/***************************************************************************
+ *   Copyright 2018 by Davide Bettio <davide@uninstall.it>                 *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU Lesser General Public License as        *
+ *   published by the Free Software Foundation; either version 2 of the    *
+ *   License, or (at your option) any later version.                       *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA .        *
+ ***************************************************************************/
+
+#include "port.h"
+#include "ccontext.h"
+#include "globalcontext.h"
+#include "mailbox.h"
+#include "socket.h"
+
+// TODO: factor the gpio and network driver code
+// so that common BEAM interface code is in libAtomVM
+#include "gpio_driver.h"
+#include "network_driver.h"
+
+
+const char *const port_ok_a = "\x2" "ok";
+const char *const port_error_a = "\x5" "error";
+
+
+term port_create_tuple2(CContext *cc, term a, term b) {
```

Comment:
just one minor thing: we are using K&R style, so I would open function braces on a new line :)

Reply:
fixed